One remaining test issue.



One test requires a certain exit code dependant on `sys.stdin.isatty()`:

```
       self.run_yq("{}", [], expect_exit_codes={0} if sys.stdin.isatty() else {2})
```

Running the tests I get

```
E           SystemExit: 0
[...]
E   AssertionError: 0 not found in {2}
```

https://github.com/kislyuk/yq/blob/v2.11.0/test/test.py#L46-L47

I'm not really sure how to tackle that.

--- a/pkgs/development/python-modules/yq/default.nix
+++ b/pkgs/development/python-modules/yq/default.nix
@@ -1,6 +1,7 @@
 { lib
-, buildPythonApplication
+, buildPythonPackage
 , fetchPypi
+, pkgs
 , argcomplete
 , pyyaml
 , xmltodict
@@ -12,11 +13,24 @@
 , toml
 }:
 
-buildPythonApplication rec {
+buildPythonPackage rec {
   pname = "yq";
   version = "2.10.1";
 
-  propagatedBuildInputs = [ pyyaml xmltodict jq argcomplete ];
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1h6nnkp53mm4spwy8nyxwvh9j6p4lxvf20j4bgjskhnhaw3jl9gn";
+  };
+
+  postPatch = ''
+    substituteInPlace test/test.py --replace "expect_exit_codes={0} if sys.stdin.isatty() else {2}" "expect_exit_codes={0}"
+  '';
+
+  propagatedBuildInputs = [
+    pyyaml
+    xmltodict
+    argcomplete
+  ];
 
   doCheck = true;
 
@@ -24,16 +38,13 @@ buildPythonApplication rec {
    pytest
    coverage
    flake8
-   jq
+   pkgs.jq
    toml
   ];
 
   checkPhase = "pytest ./test/test.py";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "1h6nnkp53mm4spwy8nyxwvh9j6p4lxvf20j4bgjskhnhaw3jl9gn";
-  };
+  pythonImportsCheck = [ "yq" ];
 
   meta = with lib; {
     description = "Command-line YAML processor - jq wrapper for YAML documents.";

--- a/pkgs/development/python-modules/yq/default.nix
+++ b/pkgs/development/python-modules/yq/default.nix
@@ -15,11 +15,11 @@
 
 buildPythonPackage rec {
   pname = "yq";
-  version = "2.10.1";
+  version = "2.11.0";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1h6nnkp53mm4spwy8nyxwvh9j6p4lxvf20j4bgjskhnhaw3jl9gn";
+    sha256 = "1gp9q5w1bjbw7wmba5hm8ippwvkind0p02n07fqa9jlqglhxhm46";
   };
 
   postPatch = ''

--- a/pkgs/development/python-modules/yq/default.nix
+++ b/pkgs/development/python-modules/yq/default.nix
@@ -10,6 +10,7 @@
 , flake8
 , jq
 , pytest
+, unixtools
 , toml
 }:
 
@@ -22,10 +23,6 @@ buildPythonPackage rec {
     sha256 = "1gp9q5w1bjbw7wmba5hm8ippwvkind0p02n07fqa9jlqglhxhm46";
   };
 
-  postPatch = ''
-    substituteInPlace test/test.py --replace "expect_exit_codes={0} if sys.stdin.isatty() else {2}" "expect_exit_codes={0}"
-  '';
-
   propagatedBuildInputs = [
     pyyaml
     xmltodict
@@ -35,6 +32,7 @@ buildPythonPackage rec {
   doCheck = true;
 
   checkInputs = [
+   unixtools.script
    pytest
    coverage
    flake8
@@ -42,7 +40,8 @@ buildPythonPackage rec {
    toml
   ];
 
-  checkPhase = "pytest ./test/test.py";
+  # tests fails if stdin is not a tty
+  checkPhase = "echo | script -c 'pytest ./test/test.py'";
 
   pythonImportsCheck = [ "yq" ];
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11798,10 +11798,7 @@ in
 
   yodl = callPackage ../development/tools/misc/yodl { };
 
-  yq = callPackage ../development/tools/yq {
-    inherit (python3Packages)
-    buildPythonApplication fetchPypi argcomplete pyyaml xmltodict pytest coverage flake8 toml;
-  };
+  yq = python3.pkgs.toPythonApplication python3.pkgs.yq;
 
   yq-go = callPackage ../development/tools/yq-go { };
 

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7614,6 +7614,8 @@ in {
 
   yowsup = callPackage ../development/python-modules/yowsup { };
 
+  yq = callPackage ../development/python-modules/yq { };
+
   yt = callPackage ../development/python-modules/yt { };
 
   yubico-client = callPackage ../development/python-modules/yubico-client { };

